pallets/xst/src/lib.rs

```diff
@@ -1087,6 +1090,58 @@ impl<T: Config> LiquiditySource<T::DEXId, T::AccountId, T::AssetId, Balance, Dis
         )
     }
 
+    fn step_quote(
+        dex_id: &T::DEXId,
+        input_asset_id: &T::AssetId,
+        output_asset_id: &T::AssetId,
+        amount: QuoteAmount<Balance>,
+        samples_count: usize,
+        deduce_fee: bool,
+    ) -> Result<VecDeque<SwapChunk<Balance>>, DispatchError> {
+        if !Self::can_exchange(dex_id, input_asset_id, output_asset_id) {
+            fail!(Error::<T>::CantExchange);
+        }
+        if amount.amount().is_zero() {
+            return Ok(VecDeque::new());
+        }
+
+        let synthetic_base_asset_id = &T::GetSyntheticBaseAssetId::get();
+        let (input_amount, output_amount, fee_amount) = if input_asset_id == synthetic_base_asset_id
+        {
+            Self::decide_sell_amounts(&input_asset_id, &output_asset_id, amount, deduce_fee, false)?
```

Comment:
In `quote`, `check_limits` is set to `true`. Why `false` here?

Did not dig into implementation of `xst` pallet; would be glad if it was clarified here (maybe as comments?)

Reply:
Got it, thanks!
I think adding short explanation in the nearby comments would help future developers; it doesn't seem immediately obvious

Reply:
thanks for adding it 👍